api/pkg/crd/migrations/seed/migrations.go

```diff
@@ -228,7 +228,9 @@ func createSecretsForCredentials(cluster *kubermaticv1.Cluster, cleanupContext *
 	if err := kubernetesprovider.CreateOrUpdateCredentialSecretForCluster(cleanupContext.ctx, cleanupContext.client, cluster); err != nil {
 		return err
 	}
-	kuberneteshelper.AddFinalizer(cluster, apiv1.CredentialsSecretsCleanupFinalizer)
+	if cluster.GetDeletionTimestamp() == nil {
+		kuberneteshelper.AddFinalizer(cluster, apiv1.CredentialsSecretsCleanupFinalizer)
+	}
```

Comment:
If I read this correctly, this will not add the finalizer when the cluster is already in deletion. That make sense.

But in line 228, right before this, we create the credentials Secrets regardless of the deletion status. Won't these secrets then be orphans once the cluster is gone?

Reply:
Ah yes, changed it to just return at the beginning of the function if there is a deletionTimestamp.
Make more sense it is even better readable this way :D